python/ray/experimental/channel/shared_memory_channel.py

```diff
@@ -323,6 +328,42 @@ def write(self, value: Any):
             )
             raise TypeError(msg) from e
 
+        # serialized_value.total_bytes *only* includes the size of the data. It does not
+        # include the size of the metadata, so we must account for the size of the
+        # metadata explicitly.
+        size = serialized_value.total_bytes + len(serialized_value.metadata)
+        if size > self._typ.buffer_size_bytes:
+            # Now make the channel backing store larger.
+            self._typ.buffer_size_bytes = size
+            # TODO(jhumphri): I assume the current writer ref object is automatically
+            # freed when the reference to it is destroyed in the line below?
+            prev_writer_ref = self._writer_ref
+            self._writer_ref = _create_channel_ref(self, self._typ.buffer_size_bytes)
+            # We need to register the new writer_ref.
+            self.ensure_registered_as_writer(force=True)
```

Comment:
Nit: resetting `self._writer_registered = False` seems a little cleaner (so future callers don't need to figure our if they need to set `force` or not).

Reply:
Done.